pages/price-feeds/use-real-time-data/aptos.mdx

```diff
@@ -1,31 +1,87 @@
+Add the Pyth Contract to your project dependencies in the `Move.toml` file:
 
-## Example Applications
+```toml copy
+[dependencies]
+Pyth = { git = "https://github.com/pyth-network/pyth-crosschain.git", subdir = "target_chains/aptos/contracts", rev = "main" }
+```
 
-- [Minimal on-chain contract](https://github.com/pyth-network/pyth-crosschain/blob/main/target_chains/aptos/examples/fetch_btc_price/sources/example.move) which updates and returns the Pyth BTC/USD price.
-- [Full-stack React app and on-chain contract](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/aptos/examples/mint_nft) which uses the [pyth-aptos-js](https://github.com/pyth-network/pyth-crosschain/tree/main/target_chains/aptos/sdk/js) package to update the price used by the contract.
-- [In-depth explanation](https://youtu.be/0b0RXi41pN0) of how Pyth works on Aptos and how to integrate Pyth data in your application from one of our contributor.
+The named addresses of `pyth`, `wormhole`, and `deployers` must be defined at compile time. These addresses are used to interact with the Pyth contract on Aptos.
 
-## Contract Addresses
+```toml copy
+[addresses]
+pyth = "0x7e783b349d3e89cf5931af376ebeadbfab855b3fa239b7ada8f5a92fbea6b387"
+deployer = "0xb31e712b26fd295357355f6845e77c888298636609e93bc9b05f0f604049f434"
+wormhole = "0x5bc11445584a763c1fa7ed39081f1b920954da14e04b32440cba863d03e19625"
+```
 
-Developers will need the address of the Pyth price feed contract on their blockchain in order to use Pyth.
-Please consult [Aptos Contract Addresses](../contract-addresses/aptos) to find the address for your blockchain.
+Consult: [Aptos Contract Addresses](../contract-addresses/aptos) for the complete list of contract addresses on different Aptos networks.
 
-## Price Feed IDs
+## Write Contract Code
 
-| Network       | Available Price Feeds                                                                                                        |
-| ------------- | ---------------------------------------------------------------------------------------------------------------------------- |
-| Aptos Testnet | [https://pyth.network/developers/price-feed-ids#aptos-testnet](https://pyth.network/developers/price-feed-ids#aptos-testnet) |
-| Aptos Mainnet | [https://pyth.network/developers/price-feed-ids#aptos-mainnet](https://pyth.network/developers/price-feed-ids#aptos-mainnet) |
+The code snippet below provides an example module fetching the BTC/USD price from Pyth price feeds:
+
+```rust {21} copy
+module example::example {
+    use pyth::pyth;
+    use pyth::price::Price;
+    use pyth::price_identifier;
+    use aptos_framework::coin;
+
+    // Fetch the pyth_update_data from Hermes and pass it to the Pyth contract to update the prices.
```

Comment:
I think this sentence should be 2nd in the paragraph. It's confusing to put this first because it makes it sound like this function is fetching the pyth_update_data, when in reality it's the off-chain code that will do this and pass it to this function.